src/mongo_odm/model.hpp

```diff
@@ -24,8 +24,51 @@
 namespace mongo_odm {
 MONGO_ODM_INLINE_NAMESPACE_BEGIN
 
+/**
+ * Helper type trait widget that helps properly forward arguments to _id constructor.
+ */
+template <typename... Ts>
+struct Widget : public std::false_type {};
+
+template <typename T, typename T2, typename... Ts>
+struct Widget<T, T2, Ts...> : public std::is_same<T, std::decay_t<T2>> {};
```

Comment:
Let's name this FirstTypeIsTheSame.  Or pick a name, but don't leave it called Widget